cou/utils/juju_utils.py

```diff
@@ -442,6 +442,21 @@ async def get_status(self) -> FullStatus:
         model = await self._get_model()
         return await model.get_status()
 
+    async def update_status(self, unit: str) -> None:
+        """Run the update_status hook on the given unit if the hook exists.
+
+        :param unit: Name of the unit to run update-status hook
+        :type unit: str
+        :raises CommandRunFailed: When update-status hook failed
+        """
+        update_status_hook = "hooks/update-status"
+        try:
+            await self.run_on_unit(unit, f"ls {update_status_hook}")
+        except CommandRunFailed:
+            logger.debug("Skipped running '%s': file does not exist", update_status_hook)
+        else:
+            await self.run_on_unit(unit, update_status_hook)
```

Comment:
Right!, thanks, I did not we can do that as well in operator framework. It seems this is the same as the directly running `hooks/update-status` except that it can run the default handler (even if the charm does not implement the update-status handler, and it does nothing internally)

Reply:
updated

Reply:
Yes, pushed